corehq/apps/userreports/pillow.py

```diff
@@ -208,6 +211,117 @@ def _save_doc_to_table(self, domain, table, doc, eval_context):
             # remove it until the next bootstrap call
             self.table_adapters_by_domain[domain].remove(table)
 
+    def _get_ucr_values(self, domain, table, doc, eval_context):
+        try:
+            return table.get_all_values(doc, eval_context)
+        except Exception as e:
+            try:
+                table.handle_exception(doc, e)
+            except UserReportsWarning:
+                # remove it until the next bootstrap call
+                self.table_adapters_by_domain[domain].remove(table)
```

Comment:
removing the table from the list might cause other tables to be skipped down on line 297:
```
l = [1,2,3]
for i in l:
   print(i)
   l.remove(i)
>> 1
>> 3
```

Reply:
I think that we don't want to remove these at all, as it could cause missing documents in the data source.

1. Change a column type, and rebuild it
2. A change comes in and errors because the data source was kept in memory and doesn't match the database definiton
3. The data source is removed here
4. Another change comes in, but there is no data source so it's not processed/no error is emitted.
5. that change never gets to the data source

Reply:
Thinking of this now, it's possible that we're open to a type of this errors now as pillows have no way to be notified that a new data source has been created. This could have been the source of bugs before.